src/snowflake/snowpark/dataframe_writer.py

```diff
@@ -131,8 +137,19 @@ def save_as_table(
                 "Unsupported table type. Expected table types: temp/temporary, transient"
             )
 
+        if column_order:
+            column_order_lower = column_order.lower()
+            if column_order_lower == "name":
```

Comment:
shouldn't we also check whether it's "append" mode (according to the docstring)?

Reply:
In "overwrite" mode, column order "name" and "index" have no impact on the result. This is why the doc says "append" mode.

Reply:
Should we raise a `ValueError` if the write mode is not `append`, and `column_order` is not default?